src/resolve/resolved-bus.c

```diff
@@ -327,6 +327,13 @@ static void bus_method_resolve_hostname_complete(DnsQuery *query) {
                 goto finish;
 
         q->bus_request = sd_bus_message_unref(q->bus_request);
+
+        if (q->manager->varlink_subscription) {
+                r = send_dns_notification(q->manager, q->answer, question);
+                if (r < 0)
+                        log_error_errno(r, "Failed to send varlink notification: %m");
+        }
+
```

Comment:
I refactored this to move the logic to dns_query_complete so it can handle all the different paths in one place.

That said, I'm not familiar with other record types at this time to determine what information should be included in such notifications.

What do you think about making this method extensible for future but not actually sending out notifications for other records with this PR? We could add a "type" field indicating the RR type. It would currently only send A and AAAA types with the hostname/ip address info, but would remain extensible for the future.